File: Frontend/src/services/tournamentService.js
import api from './api';

export const getAllTournaments = async () => {
  const response = await api.get('/tournaments');
  return response.data;
};

export const getTournamentDetails = async (id) => {
  const response = await api.get(`/tournaments/${id}`);
  return response.data;
};

export const createTournament = async (name, maxTeams, rulesDescription, creatorId) => {
  const response = await api.post('/tournaments', { name, maxTeams, rulesDescription, creatorId });
  return response.data;
};

export const registerForTournament = async (tournamentId, teamName, teamTag, userId, captainInGameName, logoUrl, captainPhoneNumber) => {
  const response = await api.post(`/tournaments/${tournamentId}/register`, { 
    teamName, 
    teamTag, 
    userId, 
    captainInGameName,
    logoUrl,
    captainPhoneNumber
  });
  return response.data;
};

export const getTournamentRegistrations = async (tournamentId, organizerUserId) => {
  const response = await api.get(`/tournaments/${tournamentId}/registrations`, { params: { organizerUserId } });
  return response.data;
};

export const approveRegistration = async (tournamentId, teamId, organizerUserId) => {
  const response = await api.put(`/tournaments/${tournamentId}/registrations/${teamId}/approve`, null, { params: { organizerUserId } });
  return response.data;
};

export const rejectRegistration = async (tournamentId, teamId, organizerUserId) => {
  const response = await api.put(`/tournaments/${tournamentId}/registrations/${teamId}/reject`, null, { params: { organizerUserId } });
  return response.data;
};

// Admin endpoints
export const getAdminTournaments = async (adminUserId) => {
  const response = await api.get('/tournaments/admin/all', { params: { adminUserId } });
  return response.data;
};

export const approveTournament = async (tournamentId, adminUserId) => {
  const response = await api.put(`/tournaments/${tournamentId}/approve-tournament`, null, { params: { adminUserId } });
  return response.data;
};

export const rejectTournament = async (tournamentId, adminUserId) => {
  const response = await api.put(`/tournaments/${tournamentId}/reject-tournament`, null, { params: { adminUserId } });
  return response.data;
};

export const updateTournamentByAdmin = async (tournamentId, data, adminUserId) => {
  const response = await api.put(`/tournaments/${tournamentId}/admin-update`, data, { params: { adminUserId } });
  return response.data;
};

export const deleteTournamentByAdmin = async (tournamentId, adminUserId) => {
  const response = await api.delete(`/tournaments/${tournamentId}/admin-delete`, { params: { adminUserId } });
  return response.data;
};

File: Backend/src/main/java/com/tournament/engine/modules/tournament/service/TournamentService.java
package com.tournament.engine.modules.tournament.service;

import com.tournament.engine.modules.tournament.dto.TournamentCreateRequest;
import com.tournament.engine.modules.tournament.dto.TournamentRegisterRequest;
import com.tournament.engine.modules.tournament.dto.TournamentResponse;

import java.util.List;

public interface TournamentService {
    List<TournamentResponse> getAllTournaments();
    TournamentResponse getTournamentDetails(Long id);
    TournamentResponse createTournament(com.tournament.engine.modules.tournament.dto.TournamentCreateRequest request);
    void registerForTournament(Long tournamentId, com.tournament.engine.modules.tournament.dto.TournamentRegisterRequest request);
    
    // Organizer endpoints
    List<com.tournament.engine.modules.tournament.dto.TournamentRegistrationResponse> getTournamentRegistrations(Long tournamentId, Long organizerUserId);
    void approveRegistration(Long tournamentId, Long teamId, Long organizerUserId);
    void rejectRegistration(Long tournamentId, Long teamId, Long organizerUserId);

    // Admin endpoints
    List<TournamentResponse> getAllTournamentsForAdmin(Long adminUserId);
    void approveTournament(Long tournamentId, Long adminUserId);
    void rejectTournament(Long tournamentId, Long adminUserId);
    void updateTournamentByAdmin(Long tournamentId, com.tournament.engine.modules.tournament.dto.TournamentCreateRequest request, Long adminUserId);
    void updateTournament(Long tournamentId, com.tournament.engine.modules.tournament.dto.TournamentCreateRequest request, Long organizerUserId);
    void deleteTournamentByAdmin(Long tournamentId, Long adminUserId);
    // Organizer & Referee management
    List<com.tournament.engine.modules.tournament.dto.TournamentOrganizerResponse> getTournamentOrganizers(Long tournamentId);
    void addTournamentOrganizer(Long tournamentId, com.tournament.engine.modules.tournament.dto.AddOrganizerRequest request, Long assignerUserId);
    void removeTournamentOrganizer(Long tournamentId, Long targetUserId, Long assignerUserId);
}
